viz_scripts/energy_calculations.ipynb

```diff
@@ -51,190 +56,103 @@
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "above-network",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "participant_ct_df = scaffolding.load_all_participant_trips(program, tq)"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "revolutionary-lounge",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "labeled_ct = scaffolding.filter_labeled_trips(participant_ct_df)"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "abroad-myanmar",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "expanded_ct = scaffolding.expand_userinputs(labeled_ct)"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "cultural-salad",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "expanded_ct = scaffolding.data_quality_check(expanded_ct)\n",
-    "expanded_ct.shape"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "improved-venture",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "## Mapping new labels with dictionaries\n",
-    "expanded_ct['Trip_purpose']= expanded_ct['purpose_confirm'].map(dic_pur)\n",
-    "expanded_ct['Mode_confirm']= expanded_ct['mode_confirm'].map(dic_re)\n",
-    "expanded_ct['Replaced_mode']= expanded_ct['replaced_mode'].map(dic_re)\n",
-    "\n",
-    "#Mapping fuel\n",
-    "expanded_ct['Mode_confirm_fuel']= expanded_ct['Mode_confirm'].map(dic_fuel)\n",
-    "expanded_ct['Replaced_mode_fuel']= expanded_ct['Replaced_mode'].map(dic_fuel)"
+    "dic_fuel = defaultdict(lambda: 'Other',dic_fuel)"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "micro-wound",
+   "cell_type": "markdown",
+   "id": "parallel-patch",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "scaffolding.unit_conversions(expanded_ct)"
+    "## Collect Data From Database"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "efficient-marking",
-   "metadata": {},
+   "id": "sufficient-kingston",
+   "metadata": {
+    "scrolled": false
+   },
    "outputs": [],
    "source": [
-    "file_suffix = scaffolding.get_file_suffix(year, month, program)\n",
-    "quality_text = scaffolding.get_quality_text(participant_ct_df, expanded_ct)"
+    "expanded_ct, file_suffix, quality_text = scaffolding.load_viz_notebook_data(year,\n",
+    "                                                                            month,\n",
+    "                                                                            program,\n",
+    "                                                                            df_ei,\n",
+    "                                                                            dic_re,\n",
+    "                                                                            dic_pur,\n",
+    "                                                                            dic_fuel)"
```

Comment:
Since the dictionaries are loaded with IPython magic outside of this function, I remove their load when not needed. To make this function itself less heavyweight, I make the dictionaries optional arguments and only if included they are used to recode the user input.

Reply:
I can do the two function solution if this Is inadequate